include/RAJA/util/View.hpp

```diff
@@ -296,6 +297,43 @@ RAJA_INLINE AtomicViewWrapper<ViewType, AtomicPolicy> make_atomic_view(
   return RAJA::AtomicViewWrapper<ViewType, AtomicPolicy>(view);
 }
 
+struct layout_left{};
+struct layout_right{};
+
+template<typename LAYOUT>
+struct Reshape;
+
+template<>
+struct Reshape<layout_right>
+{
+  template<typename T, typename...Ts>
+  static auto get(T *ptr, Ts... s)
+  {
+    constexpr int N = sizeof...(Ts);
+    return RAJA::View<T, RAJA::Layout<N, RAJA::Index_type, N-1>>(ptr, s...);
+  }
+};
+
+template<>
+struct Reshape<layout_left>
+{
+  template<typename T, typename...Ts>
+  static auto get(T *ptr, Ts... s)
+  {
+  constexpr int N = sizeof...(Ts);
+
+  std::array<RAJA::idx_t, N> extent{s...};
+
+  ///Should be a away to do this at compile time...
+  std::array<RAJA::idx_t, N> reverse_indices_array;
+  for(int i = N-1, j=0; i>-1; --i, j++) {reverse_indices_array[j] = i; }
```

Comment:
Make an array of `{s...}`, and an integer_sequence of `0...(N-1)`, pass them into a function and then you'll have all you need to do the indexing.

Reply:
@MrBurmark I just pushed up a new version, would you be able to provide some feedback?